framework/test/bundles/libH/TestBundleH.cpp

```diff
@@ -187,6 +168,50 @@ class TestBundleHServiceFactoryReturnsNullPtr : public ServiceFactory
 
 };
 
+// Simulate an error condition whereby the service factory recursively tries to get the same service for the same bundle
+class TestBundleHServiceFactoryRecursion : public ServiceFactory
+{
+public:
+
+  InterfaceMapConstPtr GetService(const Bundle& caller, const ServiceRegistrationBase& sReg)
+  {
+    static bool validRecursion = false;
```

Comment:
I will add some more comments, but I think you misread the code. A call from the framework context will cause infinite recursion. A call from a non-framework context will recurse once, by retrieving the service on behalf of the framework context. The `return nullptr` is a no-op, it is never reached.